pysb/bng.py

```diff
@@ -137,8 +137,16 @@ def _format_action_args(cls, **kwargs):
             Arguments and values to supply to BNG
         """
         if kwargs:
-            action_args = ','.join('%s=>%s' % (k, BngConsole._bng_param(v))
-                                   for k, v in kwargs.items())
+            processed_pars = []
+            for k, v in kwargs.items():
+                bng_param = BngConsole._bng_param(v)
+                if isinstance(bng_param, collections.Mapping):
+                    par_string = f'{k}' + '=>{' + ','.join(f"{k}=>{v}" for k, v in bng_param.items()) + '}'
```

Comment:
This is unwieldy and not really using the benefit of f-strings. Also re-using the local `v` variable is a bit confusing. How about something like this (not tested) :
```python
args_string = ",".join(f"{k}=>{v}" for k, v in bng_param.items())
par_string = f"{k}=>{args_string}"
```